File: src/main/java/pl/dolien/shop/feedback/FeedbackService.java
package pl.dolien.shop.feedback;

import org.springframework.data.domain.Page;
import org.springframework.security.core.Authentication;
import pl.dolien.shop.feedback.dto.FeedbackDTO;
import pl.dolien.shop.feedback.dto.FeedbackRequestDTO;
import pl.dolien.shop.feedback.dto.FeedbackResponseDTO;
import pl.dolien.shop.pagination.PaginationParams;

public interface FeedbackService {

    FeedbackDTO saveFeedback(FeedbackRequestDTO request, Authentication connectedUser);

    Page<FeedbackResponseDTO> getFeedbacksByProduct(Long productId, PaginationParams paginationParams, Authentication connectedUser);
}

File: shop-ui/src/app/feedback/services/feedback/feedback.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { PaginationParams } from '../../../pagination/models/pagination-params/pagination-params';
import { environment } from '../../../../environments/environment.development';
import { FeedbackResponse } from '../../models/feedback-response/feedback-response';
import { FeedbackRequest } from '../../models/feedback-request/feedback-request';

@Injectable({
  providedIn: 'root',
})
export class FeedbackService {
  private readonly feedbackUrl = `${environment.url}/feedbacks`;

  constructor(private httpClient: HttpClient) {}

  getFeedbacksByProduct(
    productId: number,
    paginationParams: PaginationParams
  ): Observable<FeedbackResponse> {
    const feedbacksByProductUrl = `${this.feedbackUrl}/products/${productId}?page=${paginationParams.page}&size=${paginationParams.size}`;

    return this.httpClient.get<FeedbackResponse>(feedbacksByProductUrl);
  }

  saveFeedback(request: FeedbackRequest): Observable<number> {
    return this.httpClient.post<number>(this.feedbackUrl, request);
  }
}
